### DISCUSSION TITLE
Keyball46 firmware v1.1.0

### DISCUSSION THREAD

**Discussion by koron** [Yowkees/keyball#70] [Show and tell]
Keyball46 の v1.1.0 のビルド済みファームウェアを用意しました。

種類 | 右ボール | 両ボール | 左ボール | 説明
---|---|---|---|---
**VIA対応** | [keyball_via_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331096/keyball_via_1.1.0.zip) | [keyball_via_Both_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331098/keyball_via_Both_1.1.0.zip) | [keyball_via_Left_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331099/keyball_via_Left_1.1.0.zip) | VIA (Remap) 対応
**組み立てテスト** | [keyball_test_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331101/keyball_test_1.1.0.zip) | [keyball_test_Both_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331102/keyball_test_Both_1.1.0.zip) | [keyball_test_Left_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331094/keyball_test_Left_1.1.0.zip) | テスト用キーマップ<br>RGB LEDが赤→緑→青で点灯
**デフォルト** | [keyball_default_1.1.0.zip](https://github.com/Yowkees/keyball/files/7331100/keyball_default_1.1.0.zip) | N/A | N/A | OLEDに対応。RGB LEDに一部対応

このバージョンのファームウェアではトラックボールの有無に応じて基板ごとに書き込むファームウェアが異なります。各ZIPには `_rev1_ball_` と `_rev1_noball_` の2つのファームが含まれています。`_rev1_ball_` をトラックボールのある方に、`_rev1_noball_` をトラックボールのない方に書き込んでください。

例: VIA対応右ボール版の中身

* `keyball_rev1_ball_via.hex` - トラックボールのあるPCB用
* `keyball_rev1_noball_via.hex` - トラックボールのないPCB用

## ファームウェアの選び方

通常の利用時には **VIA対応** 版を使えば十分です。

組み立てる時のテストには **組み立てテスト** 版が便利です。RGB LEDの動作確認に適した点灯パターンと、各キーのダイオードの確認に適したキーマップが含まれています。

**デフォルト** は参考用です。ご自身でソースからファームをビルドする際のテンプレートとなる `default` キーマップの動作確認ができます。

## 変更点

[前版 (v1.0.0)](https://github.com/Yowkees/keyball/discussions/46) からの変更点は以下の通りです。

* QMK 0.14.x でビルドできるようにした (See also #69)

    * VIA版: OLEDが点灯しなくなりました。代わりにRGB LEDのTwinkle以外のエフェクトが利用可能になりました
    * デフォルト版: OLEDが点灯するようになりました。RGB LEDの一部エフェクトが利用可能になりました

**Comment by Yowkees** [comment]
ファームウェアの整備まで行っていただいて、本当にありがとうございます。
早速ビルドガイドにてご紹介できるように修正させていただきます。

**Comment by freznel10** [comment]
Is there a way to program automatic shift to ball layer just like in drashna's tractyl manuform?

https://github.com/qmk/qmk_firmware/blob/master/keyboards/handwired/tractyl_manuform/5x6_right/keymaps/drashna/keymap.c#L177-L233
